src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -62,141 +61,49 @@ class CDCL(val clauses: MutableList<MutableList<Int>>, initNumber: Int = 0) {
         }.sortedBy { litIndex(it) }.filter { litIndex(it) > 0 }
 
     // values of variables
-    private val vars: MutableList<VarState> = MutableList(varsNumber + 1) { VarState(VarStatus.UNDEFINED, -1, -1) }
+    protected val vars: MutableList<VarState> = MutableList(varsNumber + 1) { VarState(VarStatus.UNDEFINED, -1, -1) }
 
     // all decisions and consequences
-    private val trail: MutableList<Int> = mutableListOf()
+    protected val trail: MutableList<Int> = mutableListOf()
 
     // decision level
-    private var level: Int = 0
+    protected var level: Int = 0
 
     // two watched literals heuristic; in watchers[i] set of clauses watched by variable i
-    private val watchers = MutableList(varsNumber + 1) { mutableSetOf<Int>() }
-    private fun litIndex(lit: Int): Int = abs(lit)
+    protected val watchers = MutableList(varsNumber + 1) { mutableSetOf<Int>() }
+    protected fun litIndex(lit: Int): Int = abs(lit)
```

Comment:
Docs?
Why this method is in "two-watched literals heuristic" block? Looks like a general-purpose method. (actually, it is even not a method, so you could make it a top-level function or static method)